From 00398f1d2c254cd74afc474cf3a4ef4ecceef9d0 Mon Sep 17 00:00:00 2001
From: Nadav Haklai <nadavh@marvell.com>
Date: Tue, 29 Mar 2016 17:40:30 +0300
Subject: [PATCH 089/239] mv_ddr: apn806: Fix tip interface write func

This patch fixes incorrect param passed to the internally
called if read func. No need to add dunit base addr prior to
the if read func call, because it adds it internally.

Change-Id: Ibe7f9cc03605c9ac0dd23275838fd4921fd9859e
Signed-off-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28641
Reviewed-on: http://vgitil04.il.marvell.com:8080/28796
Tested-by: Star_Automation <star@marvell.com>
---
 drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.c b/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.c
index d26473b..f6ac68a 100644
--- a/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.c
+++ b/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.c
@@ -171,8 +171,6 @@ static int ddr3_tip_apn806_if_write(u8 dev_num, enum hws_access_type interface_a
 {
 	u32 ui_data_read;
 
-	reg_addr += TIP_BASE_ADDRESS;
-
 	if (mask != MASK_ALL_BITS) {
 		CHECK_STATUS(ddr3_tip_apn806_if_read
 			     (dev_num, ACCESS_TYPE_UNICAST, if_id, reg_addr,
@@ -180,6 +178,8 @@ static int ddr3_tip_apn806_if_write(u8 dev_num, enum hws_access_type interface_a
 		data_value = (ui_data_read & (~mask)) | (data_value & mask);
 	}
 
+	reg_addr += TIP_BASE_ADDRESS;
+
 	reg_write(reg_addr, data_value);
 
 	return MV_OK;
-- 
1.9.1

